components/add_users_to_group_modal/add_users_to_group_modal.tsx

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState} from 'react';
+
+import {Modal} from 'react-bootstrap';
+
+import {FormattedMessage} from 'react-intl';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import {localizeMessage} from 'utils/utils.jsx';
+import {Group} from 'mattermost-redux/types/groups';
+
+import 'components/user_groups_modal/user_groups_modal.scss';
+import {ModalData} from 'types/actions';
+import AddUserToGroupMultiSelect from 'components/add_user_to_group_multiselect';
+import {ActionResult} from 'mattermost-redux/types/actions';
+
+import LocalizedIcon from 'components/localized_icon';
+import {t} from 'utils/i18n';
+
+export type Props = {
+    onExited: () => void;
+    groupId: string;
+    group: Group;
+    backButtonCallback: () => void;
+    actions: {
+        addUsersToGroup: (groupId: string, userIds: string[]) => Promise<ActionResult>;
+        openModal: <P>(modalData: ModalData<P>) => void;
+    };
+}
+
+const AddUsersToGroupModal = (props: Props) => {
+    const [show, setShow] = useState(true);
+    const [usersToAdd, setUsersToAdd] = useState<UserProfile[]>([]);
+    const [showUnknownError, setShowUnknownError] = useState(false);
+
+    const doHide = () => {
+        setShow(false);
+    };
+
+    const isSaveEnabled = () => {
+        return usersToAdd.length > 0;
+    };
+
+    const addUserCallback = (users: UserProfile[]): void => {
+        setUsersToAdd(users);
+    };
+
+    const deleteUserCallback = (users: UserProfile[]): void => {
+        setUsersToAdd(users);
+    };
+
+    const addUsersToGroup = async (users?: UserProfile[]) => {
+        if (!users || users.length === 0) {
+            return;
+        }
+        const userIds = users.map((user) => {
+            return user.id;
+        });
+
+        const data = await props.actions.addUsersToGroup(props.groupId, userIds);
+
+        if (data?.error) {
+            setShowUnknownError(true);
+        } else {
+            goBack();
+        }
+    };
+
+    const goBack = () => {
+        props.backButtonCallback();
+        props.onExited();
+    };
+
+    return (
+        <Modal
+            dialogClassName='a11y__modal user-groups-modal-create'
+            show={show}
+            onHide={doHide}
+            onExited={props.onExited}
+            role='dialog'
+            aria-labelledby='createUserGroupsModalLabel'
+            id='addUsersToGroupsModal'
+        >
+            <Modal.Header closeButton={true}>
+                <button
+                    type='button'
+                    className='modal-header-back-button btn-icon'
+                    aria-label='Close'
+                    onClick={() => {
```

Comment:
This anonymous function makes no sense, we can pass just `goBack` directly and has exactly the same effect, and also, if we make the `goBack` function not changing every single render, we can avoid re-rendering this button in every render too.